Update docs for SEMGREP_REPO_NAME and SEMGREP_REPO_DISPLAY_NAME (#1472)
semgrep/semgrep-docs: docs/semgrep-ci/ci-environment-variables.md

### `SEMGREP_REPO_NAME`

> Set `SEMGREP_REPO_NAME` to create a repository name when scanning with a CI provider we don't provide support for. For hyperlinks and PR comments to work, this name should be the same as your repository name understood by your CI provider.

Reviewer: ```suggestion
Set `SEMGREP_REPO_NAME` to create a repository name when scanning with a [CI provider that Semgrep doesn't provide explicit support for](/deployment/add-semgrep-to-other-ci-providers/). For hyperlinks and PR comments to work, this name should be the same as your repository name understood by your CI provider.
```

> To avoid hardcoding this value, check your CI provider's documentation for available environment variables that can automatically detect the correct values for every CI job. 

Reviewer 2: Should we add that Semgrep automatically sets it for most common CI providers?

> 

Reviewer: ```suggestion

Semgrep automatically detects `SEMGREP_REPO_NAME` if your [provider is listed in Semgrep Cloud Platform](/deployment/add-semgrep-to-ci/#guided-setup-for-ci-providers-in-scp). In this case, there is no need to set the variable.
```
Hi @emjin! You can choose to add this line per @bkettle's suggestion if you so choose. It's a little wordy, but that being said, sometimes people have a hard time discerning which env vars to set so I tend to err on spelling things out. I'd say 30-40% of the community questions are about CI providers.)

```

### `SEMGREP_REPO_DISPLAY_NAME`

> Set `SEMGREP_REPO_DISPLAY_NAME` to define the name used for the project associated with a given Semgrep scan. By default, `SEMGREP_REPO_DISPLAY_NAME` has the same value as `SEMGREP_REPO_NAME`. This allows you to use a different name for your project than the repository name or to scan a monorepo as multiple projects.

Reviewer: A 'scan' sounds a little ambiguous? It could mean a discrete, specific scan that has been run, it could be a CLI scan, etc. Setting `SEMGREP_REPO_DISPLAY_NAME` essentially creates a separate project entry, so maybe we can say:
```suggestion
Set `SEMGREP_REPO_DISPLAY_NAME` to define the name used for the project in Semgrep Cloud Platform. By default, `SEMGREP_REPO_DISPLAY_NAME` has the same value as `SEMGREP_REPO_NAME`. This allows you to use a different name for your project than the repository name or to scan a monorepo as multiple projects.
```

> #### Scanning a monorepo as multiple projects

Reviewer: ```suggestion
#### Scan a monorepo as multiple projects
```
We use simple present these days for the header, mostly to be clear when something is a verb and something is a noun.
